examples/Overhead.hs

```diff
@@ -11,6 +11,7 @@ import GHC.Stats as GHC
 
 main :: IO ()
 main = do
+  M.initializeTime -- Need to do this before calling M.getTime
```

Comment:
I'm confused as to why this explicit call to `initializeTime` is necessary. Doesn't `defaultMain` call it already?

Reply:
Yes, but it's not documented that `defaultMain` calls `initializeTime`. I think this example shows correct precautionary use.

Reply:
Still, it feels strange that we'd have a call to `initializeTime` for no benefit. Perhaps we should at least document in `initializeTime` itself that several functions call this before running measurements.

As things stand, I'm inclined to remove this use of `initializeTime`, as it only serves to baffle users.

Reply:
I agree that it couldn't hurt to mention that in the docs of `initializeTime`. The primary reasoning I have for including this line here is

1. I figure that many potential users of Criterion will learn from the provided examples and use the docs as necessary. At least that's how I start using a new library with a large API. I think it's conceivable that someone would see a call to `getTime` here and use it elsewhere without realizing the significance of `defaultMain`.

2. While I don't mind adding the information in the docs, I don't think users should have to know the implementation details of a function in order to see if a program is correct. That is to say, it wouldn't be obvious to me that this module is bug-free unless I already knew that `defaultMain` calls `initializeTime`.

3. The overhead that a single call to `initializeTime` incurs is negligible in almost any circumstance

4. Leaving out call to `initializeTime` results in behavior that's odd and hard to debug if you don't already know what you're looking for (as I have learned in the past week or so). So erring on the side of caution might save a lot of time for users.

So all things considered, I think having this here would be better for potential users of Criterion. If you still feel strongly against this, I'll remove it anyway and add the docs.

Reply:
Fair enough.

I'm still not quite happy with the current state of the Haddocks for `initializeTime` and `getTime`, but I'll tackle that issue separately (see #197).